src/Akka.Persistence.Sql.Linq2Db/Journal/DAO/BaseByteArrayJournalDao.cs

```diff
@@ -390,32 +392,34 @@ public override
                 }
 
                 
-                //TODO: Is there a better way to do this async?
-                //return Source
-                //    .FromObservable(query.AsAsyncEnumerable().ToObservable())
-                
-                return Source.From(await query.ToListAsync())
+                return AsyncSource<JournalRow>.FromEnumerable(query,async q=>await q.ToListAsync())
```

Comment:
The idea with this change is that Streams should be 'lazy'. While in most cases this specific method is safely called from inside another stream's stage, keeping things consistent seemed worth it.